packages/apputils/src/mainareawidget.ts

```diff
@@ -164,59 +164,54 @@ export class MainAreaWidget<T extends Widget = Widget>
   }
 
   /**
-   * Handle `'activate-request'` messages.
+   * Handle the DOM events for the main area widget.
+   *
+   * @param event - The DOM event sent to the class.
+   *
+   * #### Notes
+   * This method implements the DOM `EventListener` interface and is
+   * called in response to events on the widget's DOM node. It should
+   * not be called directly by user code.
    */
-  protected onActivateRequest(msg: Message): void {
-    if (this._isRevealed) {
-      if (this._content) {
-        this._focusContent();
-      }
-    } else {
-      this._spinner.node.focus();
+  handleEvent(event: Event): void {
+    switch (event.type) {
+      case 'mousedown':
+        // Focus content in capture phase to ensure relevant
+        // commands operate on the current main area widget.
+        if (!this.node.contains(document.activeElement)) {
```

Comment:
You were catching everything on the document. That is more events than is necessary. I am only catching events on `this.node`.

I will update the PR to not use `handleEvent` because you're right about sub-classes. There are only two spots in JupyterLab where a sub-class has to think about `super.handleEvent(...)`:
- In a debugger dialog: https://github.com/jupyterlab/jupyterlab/blob/master/packages/debugger/src/dialogs/evaluate.ts#L85
- In a `SplitPanel` that we should deprecate by adding this functionality in Lumino, finally: https://github.com/jupyterlab/jupyterlab/blob/master/packages/settingeditor/src/splitpanel.ts#L20

Reply:
Thanks @afshin

Indeed my intent was to listen on the node - thanks for correcting that.